libs/llm/util.ts

```diff
@@ -0,0 +1,25 @@
+import { Logger } from '@nestjs/common';
+
+const logger = new Logger('llm.util');
+
+export const parseJSON = <T = any>(response: string): T | null => {
+  if (!response) return null;
+
+  try {
+    // handle ```json
+    if (response.startsWith('```')) {
+      response = response.substring(3);
```

Comment:
Can't you just skip the first and the last lines? 
Finding the first and last indexes of '\n' and trimming the response accordingly.